src/TaskManager.cs

```diff
@@ -222,7 +220,44 @@ public ActiveTask CreateTask(UserParameter parameter)
             }
         }
 
-        public SessionInfo GetSession(SerConnection connection, ActiveTask task)
+        public IDoc GetSessionAppConnection(Uri uri, Cookie cookie, string appId)
+        {
+            try
+            {
+                var url = ServerUtils.MakeWebSocketFromHttp(uri);
+                var connId = Guid.NewGuid().ToString();
+                url = $"{url}/app/engineData/identity/{connId}";
```

Comment:
why you open first global and than with global.OpenDocAsync the app.
better is to direct open /app/{APPID}/identity/{connID}

Reply:
@mossbeard please give me here feedback